src/api/CMakeLists.txt

```diff
@@ -32,23 +37,32 @@ set(TSAPI_PUBLIC_HEADERS
 target_link_libraries(tsapi PRIVATE libswoc yaml-cpp::yaml-cpp PCRE::PCRE OpenSSL::SSL)
 set_target_properties(tsapi PROPERTIES PUBLIC_HEADER "${TSAPI_PUBLIC_HEADERS}")
 
+# Items common between api and other ts libraries
+add_library(tsapicommon
+  ConfigUpdateCbTable.cc # configmanager
+  InkContInternal.cc # proxy, http
+  InkVConnInternal.cc # proxy, http
+  APIHook.cc # iocore
+  APIHooks.cc # proxy
+  LifecycleAPIHooks.cc # proxy, http
+  HttpHookState.cc # proxy, http
+)
+add_library(ts::tsapicommon ALIAS tsapicommon)
+target_link_libraries(tsapicommon
+  PUBLIC ts::inknet ts::http
+  PRIVATE ts::proxy ts::hdrs ts::tscore)
```

Comment:
Just to record it here: we decided in the PR/issue meeting to rename this to something other than `tsapicommon`. Maybe `tsapibackend`?